website/docs/help-and-support/troubleshooting-guide/widget-errors.md

```diff
@@ -1,38 +1,84 @@
 
-:::info
-If you still can't see a JSON form and need help debugging an error, please get in touch with us on our [Discord Server](https://discord.com/invite/rBTTVJp) or email us at support@appsmith.com.
-:::
\ No newline at end of file
+## Query/API/JS object execution errors
+You may see below errors when executing an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) using a widget property.
+
+### Sync fields can't execute async framework actions
+
+You may see this error when binding an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS Object](../../core-concepts/writing-code/javascript-editor-beta/) execution to a widget property that expects data, and can't be used to trigger an action. 
+
+#### Error message
+You would see one of the below error messages based on API/Query or a JS Object function execution.
+
+> Found a reference to {{action}} during evaluation. Sync fields cannot execute async framework actions. Please remove any direct/indirect references to {{action}} and try again.
+
+ or 
+
+> Found a Promise() during evaluation. Sync fields cannot execute asynchronous code.
+
+
+**To-Do to add screenshot**
+
+#### Cause
+**Action** refers to the execution of an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/). 
+
+Action can only be performed by binding it to an [async](../../core-concepts/writing-code/workflows#async-fields) field. When you bind an action to a [sync](../../core-concepts/writing-code/workflows#sync-fields) field that only expects data it throws an error.
+
+For example, you are executing a `storeValue()` function in a `TableData` property of a table. The 'TableData` property expects data and can't execute a function, so throws an error.
+
+> Found a reference to `storeValue` during evaluation. Sync fields can't execute async framework actions. 
+Please remove any direct/indirect references to `storeValue` and try again.
+
+Similarly, if you try to execute a JS Object function `<JSOBJECT_NAME.FUNCTION_NAME>` in the `TableData` property, it again throws an error:
+
+> Found a Promise() during evaluation. Sync fields cannot execute asynchronous code
+
+#### Solution
+You would have to invoke the [`.data` property](../../core-concepts/writing-code/workflows#use-case) of an API/query/JS object. 
+
+For example, you have an API ```getFullNameOfLoggedInUser``` that returns a user's full name, and you wish to add the user's name based on the logged-in user to show a welcome text, `Welcome! <LOGGED_IN_USER_NAME>`. 
```

Comment:
In this example, let's put a generic name for the API like  `Api1`. Also we don't need to make the user further imagine that this API returns full name of logged in user and all that.  So suggestion to reduce and combine the two lines to 

For example, if you have an API named ```Api1```, you could bind the response generated by the API to a Text widget by referencing the `.data` property as shown in the code snippet below.
```
{{Api1.data}}
```

Also I strongly feel this should be a JSObject example and not with API